lib/api/core/statistics.js

```diff
@@ -145,6 +145,55 @@ module.exports = function (kuzzle) {
   };
 
   /**
+   * Gets the lasts saved statistics frame from a date
+   *
+   * @param requestObject
+   * @returns {Promise} promise resolved as a ResponseObject
+   */
+  this.getStats = function(requestObject) {
+    var
+      deferred = q.defer(),
+      response = {
+        statistics: {}
+      },
+      frames,
+      currentDate,
+      since;
+
+    currentDate = new Date().getTime();
+    since = requestObject.data.body.since;
+    if (since > currentDate) {
+      deferred.resolve(new ResponseObject(requestObject, response));
+      return deferred.promise;
+    }
+    if (!this.lastFrame) {
+      if (!since || currentDate >= new Date(since).getTime()) {
+        response.statistics[currentDate] = this.currentStats;
+      }
+      deferred.resolve(new ResponseObject(requestObject, response));
+      return deferred.promise;
+    }
+
+    this.kuzzle.services.list.statsCache.getAllKeys()
+      .then(keys => {
+        frames = keys;
+        // Statistics keys are timestamp. Ordering them guarantees stats frames to be returned in the right order
+        return this.kuzzle.services.list.statsCache.mget(keys.sort());
+      })
+      .then(values => {
+        values.forEach((v, idx) => {
+          if (!since || (new Date(since).getTime()) <= (new Date(Number(frames[idx])).getTime())) {
```

Comment:
``new Date(since).getTime()`` should only be calculated once instead of at each iteration.

Moreover, since this argument is provided by users, it should also be tested against the NaN value (the result of a problem while trying to converting the argument to a Epoch value) to return an empty stats result instead of all results like it does in this current implementation.

I think that putting this test line 165 along with the one testing against a timestamp in the future would be the right place.